### PR TITLE
[improve][client] Add schema typing to consumer and producer

### PR DESCRIPTION
**Pull request by romainbrancourt**
### Modifications
Add schema typing to consumer and producer

### Verifying this change

- [ ] Make sure that the change passes the CI checks.

This change is a trivial rework / code cleanup without any test coverage.

### Documentation

- [ ] `doc-required` 
(Your PR needs to update docs and you will update later)

- [x] `doc-not-needed` 
(Please explain why)

- [ ] `doc` 
(Your PR contains doc changes)

- [ ] `doc-complete`
(Docs have been already added)

### CODE REVIEW

**Review by shibd — APPROVED** [review]

**Review by RobertIndie — COMMENTED** [review]
Could you add some schema tests to verify this PR?

**Comment by romainbrancourt** [comment]
@RobertIndie 
I added a test, do you think it's enough or you want me to create a test for each schemaType ?

File: index.d.ts [hunk 1]
```diff
@@ -236,3 +245,21 @@ export type ConsumerCryptoFailureAction =
   'FAIL' |
   'DISCARD' |
   'CONSUME';
+
+  export type SchemaType =
```
Comment (k2la) [inline index.d.ts:249]:
```suggestion
export type SchemaType =
```

**Review by k2la — CHANGES_REQUESTED** [review]

**Comment by k2la** [comment]
@romainbrancourt 
I think this can be better if you add tests about `consumer`.

**Comment by k2la** [comment]
LGTM

**Pull request merged**